File: backend/temp/solution-two-sum.js
const fs = require('fs');

function main() {
    let data;
    try {
        data = fs.readFileSync('input.txt', 'utf8');
    } catch (err) {
        console.error("Failed to open input.txt");
        return 1;
    }

    const lines = data.trim().split('\n');

    // First line: array of numbers
    const nums = lines[0].trim().split(/\s+/).map(Number);

    // Second line: target number
    const target = parseInt(lines[1], 10);

    // Solve using a hash map
    const numMap = new Map();

    for (let i = 0; i < nums.length; i++) {
        const complement = target - nums[i];
        if (numMap.has(complement)) {
            console.log(numMap.get(complement), i);
            return 0;
        }
        numMap.set(nums[i], i);
    }

    return 0;
}

main();

File: backend/temp/solution-two-sum.py
def main():
    try:
        with open("input.txt", "r") as file:
            # Read the first line and parse numbers
            line = file.readline()
            nums = list(map(int, line.strip().split()))

            # Read the target value
            target_line = file.readline()
            target = int(target_line.strip())

    except FileNotFoundError:
        print("Failed to open input.txt")
        return 1
    except Exception as e:
        print(f"Error reading input.txt: {e}")
        return 1

    # Solve using a hash map (dictionary)
    num_map = {}
    for i, num in enumerate(nums):
        complement = target - num
        if complement in num_map:
            print(num_map[complement], i)
            return 0
        num_map[num] = i

    return 0

if __name__ == "__main__":
    main()
